react-front-end/__tests__/tsrc/search/components/GallerySearchResult.test.tsx

```diff
@@ -120,5 +134,26 @@ describe("<GallerySearchResult />", () => {
         });
       }
     );
+
+    it.each([
+      ["item", languageStrings.searchpage.gallerySearchResult.viewItem],
+      ["attachment", languageStrings.searchpage.gallerySearchResult.ariaLabel],
+    ])(
+      "shows a dialog to list DRM violations for %s",
+      async (_: string, galleryEntryLabel: string) => {
+        const { getByLabelText } = await renderGallery([
+          galleryDrmUnauthorisedItem,
+        ]);
+        await act(async () => {
+          await userEvent.click(getByLabelText(galleryEntryLabel));
+        });
```

Comment:
There's a lot of boilerplate here. I wonder if we need a general (available for all tests) function like:

```typescript
const asyncClick = (element: Element, init?: MouseEventInit, {skipHover, clickCount}?: clickOptions): void =>
  act(async () => {
    await userEvent.click(element);
  });
```

And then we'd just be able to have:

```suggestion
        await asyncClick(getByLabelText(galleryEntryLabel));
```

I wonder too if `asyncClick` is the best name. Maybe `clickThenWait` as really this is all about clicking a button and then waiting for React to do it's state updates etc. Hmmm... :thinking:

Reply:
As discussed, we leave this for now and try to do it maybe in next tech debt.